planner/core/show_predicate_extractor.go

```diff
@@ -95,3 +97,47 @@ func (e *ShowColumnsTableExtractor) explainInfo() string {
 	}
 	return s
 }
+
+// ShowTablesTableExtractor is used to extract some predicates of tables.
+type ShowTablesTableExtractor struct {
+	ShowColumnsTableExtractor
+}
+
+// Extract implements the ShowTablesTableExtractor Extract interface
+func (e *ShowTablesTableExtractor) Extract(show *ast.ShowStmt) bool {
+	if show.Pattern != nil && show.Pattern.Pattern != nil {
+		pattern := show.Pattern
+		switch pattern.Pattern.(type) {
+		case *driver.ValueExpr:
+			// It is used in `SHOW TABLE in t LIKE `abc``.
+			ptn := pattern.Pattern.(*driver.ValueExpr).GetString()
+			patValue, patTypes := stringutil.CompilePattern(ptn, pattern.Escape)
+			if !collate.NewCollationEnabled() && stringutil.IsExactMatch(patTypes) {
```

Comment:
I don't quite understand. Why do you need this logic? Please help to explain it.

Reply:
It is copied from planner/core/expression_rewriter.go:1634. 

```
Treat predicate 'like' the same way as predicate '=' when it is an exact match and new collation is not enabled.
```

Reply:
Could remove this `if` statement, then use `CompileLike2Regexp` directly?

Reply:
OK, I will remove ```NewCollationEnabled``` condition. but ```IsExactMatch``` still retained. ```Equal``` is faster than ```regexp```.

Reply:
Got it.